### PR TITLE
use stock Django auth views

### PR DESCRIPTION
**Pull request by uraniumanchor**
# Contributing to the Donation Tracker

- [x] I've added tests or modified existing tests for the change.
- [x] I've humanly end-to-end tested the change by running an instance of the tracker.

### Issue from Pivotal Tracker

https://www.pivotaltracker.com/story/show/170791654

### Description of the Change

Django 1.11 improved the stock user/auth views so a lot of the custom wrappers we had around them are no longer necessary.

Also, the changed password reset token method seems a lot more robust, so I borrowed the technique for the user registration flow, as well. Some failures that would render errors before just return 403s instead, I don't think anybody is going to stumble onto a token error by accident.

There's a handful of changes to make CI work with the new `.env` flow, as well.

### Verification Process

Went through the user flow, made sure it still worked (verified the urls that the emails contain), went through the reset flow as well, did the same.

### CODE REVIEW

File: forms.py [hunk 1]
```diff
@@ -933,16 +881,14 @@ def check_token(self):
 
     def clean_username(self):
         AuthUser = get_user_model()
-        usersWithName = AuthUser.objects.filter(
+        existing = AuthUser.objects.filter(
             username__iexact=self.cleaned_data['username']
```
Comment (yunaa) [inline forms.py]: Do we want `normalize_username` somewhere in here?

File: views/auth.py [hunk 2]
```diff
@@ -163,44 +34,59 @@ def register(request):
     )
 
 
+INTERNAL_REGISTRATION_URL_TOKEN = 'register-user'
+INTERNAL_REGISTRATION_SESSION_TOKEN = '_register_user_token'
```
Comment (yunaa) [inline views/auth.py]: Can we call this a `KEY`, since it is used as a session key?

Comment (yunaa) [reply views/auth.py]: Now that I'm reading more, I must be misunderstanding this pattern. Where can I read more about it?

File: forms.py (hunk 1 above)
Comment (uraniumanchor) [reply forms.py]: Didn't even know that was a thing, but I like that.

File: views/auth.py (hunk 2 above)
Comment (uraniumanchor) [reply views/auth.py]: It's borrowing from the Django implementation pretty heavily here, but I suppose there's no reason to stick to it THAT hard.

Comment (yunaa) [reply views/auth.py]: I don't mind, I was just suffering from confusion.

File: forms.py (hunk 1 above)
Comment (uraniumanchor) [reply forms.py]: With the changes I'm about to push I just tried `ﬁ` (one character, uFB01) and it correctly changed it to `fi`, (two characters, u0066 and u0069).

File: views/auth.py (hunk 2 above)
Comment (uraniumanchor) [reply views/auth.py]: Too late, I changed it anyway.

File: forms.py (hunk 1 above)
Comment (uraniumanchor) [reply forms.py]: Actually I should just write a test for that. I swear I'm awake.

**Pull request merged**